### PR TITLE
[spirv] Support static struct member functions

### PR DESCRIPTION
**Pull request by antiagainst**
These functions are just translated into standalone functions
through the normal translation path.

### CODE REVIEW

**Comment by antiagainst** [comment]
@ehsannas: PTAL

File: tools/clang/test/CodeGenSPIRV/method.struct.method.hlsl [hunk 1]
```diff
@@ -48,51 +59,63 @@ struct T {
...
+// CHECK:          {{%\d+}} = OpFunctionCall %float %T_fn_nested %t
+// CHECK:          {{%\d+}} = OpFunctionCall %float %S_fn_static
+// CHECK:          {{%\d+}} = OpFunctionCall %float %T_fn_static
 // CHECK:                     OpFunctionEnd
 float main() : A {
     S s;
     T t;
-    return s.fn_no_ref() + s.fn_ref() + s.fn_call(5.0) + t.fn_nested();
+    return s.fn_no_ref() + s.fn_ref() + s.fn_call(5.0) + t.fn_nested() +
+           s.fn_static() + t.fn_static();
```
Comment (ehsannas) [inline tools/clang/test/CodeGenSPIRV/method.struct.method.hlsl]: Would you also add ``S::fn_static()`` or ``T::fn_static()`` to the test?

Comment (antiagainst) [reply tools/clang/test/CodeGenSPIRV/method.struct.method.hlsl]: Done

**Comment by antiagainst** [comment]
It seems the build succeeded but the status didn't get conveyed back to GitHub: https://ci.appveyor.com/project/marcelolr/directxshadercompiler/build/1.0.958

Can we restart a build job for this pull request? Thanks!

**Comment by marcelolr** [comment]
Build restarted ...

**Comment by antiagainst** [comment]
Thanks @marcelolr!

**Pull request merged**